lib/active_admin/csv_builder.rb

```diff
@@ -32,7 +32,9 @@ def self.default_for_resource(resource)
 
     def initialize(options = {}, &block)
       @resource = options.delete(:resource)
-      @columns, @options, @block = [], options, block
+      @columns = []
+      @options = ActiveAdmin.application.csv_options.merge options
+      @block = block
```

Comment:
I introduced the change to make this assignment sequential instead of parallel separately in #5803, so I would need you to rebase on top of that.